src/msg/async/AsyncMessenger.cc

```diff
@@ -396,6 +396,7 @@ int AsyncMessenger::client_bind(const entity_addr_t &bind_addr)
   lock.Lock();
   if (did_bind) {
     assert(my_inst.addr == bind_addr);
+    lock.Unlock();
```

Comment:
nit, better off using `Mutex::Locker`. or instead change to `std::mutex`, and use `std::lock_guard` instead.